src/tests/profiler/native/gcbasicprofiler/gcbasicprofiler.cpp

```diff
@@ -80,43 +77,11 @@ HRESULT GCBasicProfiler::GarbageCollectionStarted(int cGenerations, BOOL generat
             return S_OK;
         }
 
-        // Assuming an initial gc heap like this.
```

Comment:
as discussed today please check with profiling folks if there's a way you can check for segments vs regions since this is perfectly valid for segments. CC @davmason.

Reply:
Could we infer whether it's segments by checking the size of the ranges returned from GetGenerationBounds? From my understanding if regions are enabled we should see 4 mb segments, right?

We could add an API to query it too, I'm not against that but I'm having trouble coming up with a scenario that profilers would need to know about regions versus segments.

Reply:
as public functionalities go, the profiler doesn't need to know about regions vs segs but this code is doing internal validation as it knows that there's a generation start object that we allocate with segs which does not exist with regions.

segments could be 4mb too (you can set it with a config). I was mainly wondering if there's a way to check in the profiler if clrgc.dll is loaded or something (since regions only exist in clrgc.dll, not coreclr.dll for this release). 

another way, if we want to keep some internal validation, is to check if objectRanges[1].rangeStart and objectRanges[2].rangeStart are 0x18 bytes apart, and if so then gen1 should also be 0x18 bytes. and if those are not 0x18 bytes apart then we know we are using regions.

Reply:
> I was mainly wondering if there's a way to check in the profiler if clrgc.dll is loaded or something (since regions only exist in clrgc.dll, not coreclr.dll for this release).

There's no profiler specific way to look at the native libraries loaded in a process, but each OS exposes ways native code can do it. The MetadataGetDispenser test does almost what you want when it looks for coreclr: https://github.com/dotnet/runtime/blob/eee7a19541407e0036e0bdf2f7c42046651556b4/src/tests/profiler/native/metadatagetdispenser/metadatagetdispenser.cpp#L130-L250

The only major difference is it counts on coreclr.dll already being loaded for the Windows path, so if you choose to go this route you should call [EnumProcessModules](https://docs.microsoft.com/en-us/windows/win32/api/psapi/nf-psapi-enumprocessmodules) instead of LoadLibrary on Windows.

Reply:
thanks @davmason. in that case I think it'd probably be better to just do the other thing I mentioned which is simpler and also fulfills the purpose of doing a bit of internal validation. what do you think @cshung?